src/transform/predicate_pushdown.rs

```diff
@@ -74,6 +74,38 @@ impl crate::Transform for PredicatePushdown {
 }
 
 impl PredicatePushdown {
+    fn is_safe_leaf(s: &ScalarExpr, input_arity: usize) -> bool {
+        match s {
+            ScalarExpr::Column(c) => *c < input_arity,
+            ScalarExpr::Literal(_, _) => true,
+            _ => false,
+        }
+    }
+
+    /// Defines a criteria for inlining scalar expressions.  A scalar expression is "safe" to
+    /// inline if it only refers to non-computed columns, and:
+    /// * It's a leaf, or
+    /// * it's a unary function of a safe expression, or
+    /// * it's a binary function having one argument be safe and one argument be a leaf.
```

Comment:
Style stuff but: these comments are just the method written in English. It would be super-helpful to have an explanation of what "safe" means, though. It seems to be something strictly between "performs no additional computation" and "prevents exponential blow-up". Like, this will inline a `UnaryFunc` with a `Regex` match five times if it is used five times in the predicate, right?